doc/release/1.5.0-notes.rst

```diff
@@ -39,39 +57,230 @@ New features
 
 `scipy.io` improvements
 -----------------------
+`scipy.io.wavfile` error messages are more explicit about what's wrong, and 
+extraneous bytes at the ends of files are ignored instead of raising an error 
+when the data has successfully been read.
 
+`scipy.io.loadmat` gained a ``simplify_cells`` parameter, which if set to 
+``True`` simplifies the structure of the return value if the ``.mat`` file 
+contains cell arrays.
+
+``pathlib.Path`` objects are now supported in `scipy.io.mmio` functions
 
 `scipy.linalg` improvements
 ---------------------------
-
+`scipy.linalg.eigh` has been improved. Now various ``LAPACK`` drivers can be 
+selected at will and also subsets of eigenvalues can be requested via 
+``subset_by_value`` keyword. Another keyword ``subset_by_index`` is introduced.
+Keywords ``turbo`` and ``eigvals`` are deprecated.
+
+Similarly, standard and generalized Hermitian eigenvalue ``LAPACK`` routines 
+``?<sy/he>evx`` are added and existing ones now have full ``_lwork``
+counterparts.
+
+Wrappers for the following ``LAPACK`` routines have been added to 
+`scipy.linalg.lapack`:
+
+- ``?getc2``: computes the LU factorization of a general matrix with complete 
+pivoting
+- ``?gesc2``: solves a linear system given an LU factorization from ``?getc2``
+- ``?gejsv``: computes the singular value decomposition of a general matrix 
+with higher accuracy calculation of tiny singular values and their 
+corresponding singular vectors
+- ``?geqrfp``: computes the QR factorization of a general matrix with 
+non-negative elements on the diagonal of R
+- ``?gtsvx``: solves a linear system with general tridiagonal matrix
+- ``?gttrf``: computes the LU factorization of a tridiagonal matrix
+- ``?gttrs``: solves a linear system given an LU factorization from ``?gttrf``
+- ``?ptsvx``: solves a linear system with symmetric positive definite 
+tridiagonal matrix
+- ``?pttrf``: computes the LU factorization of a symmetric positive definite 
+tridiagonal matrix
+- ``?pttrs``: solves a linear system given an LU factorization from ``?pttrf``
+- ``?pteqr``: computes the eigenvectors and eigenvalues of a positive definite 
+tridiagonal matrix
+- ``?tbtrs``: solves a linear system with a triangular banded matrix
+- ``?csd``: computes the Cosine Sine decomposition of an orthogonal/unitary 
+matrix
+
+Generalized QR factorization routines (``?geqrf``) now have full ``_lwork`` 
+counterparts.
+
+`scipy.linalg.cossin` Cosine Sine decomposition of unitary matrices has been 
```

Comment:
we had this discussion but we couldn't came up with anything better and it is kind of what everybody goes with. So we counted on the fact that it is under linalg namespace. But yes I would really appreciate a better one.

Reply:
either `csd` (in analogy with `svd`) or a more explicit `cossin_decomp` would be better imho

Reply:
`csd` is already overloaded in [`scipy.signal.csd`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.csd.html) and some circles already use it for a shortcut to cumulative power spectral density. So that won't fly

Frankly, `cossin_decomp` is even uglier than `cossin` also becomes a one-off since the others don't have it. 

Why I prefer `cossin` is that there is nothing called `cossin` in any trig libraries. And you can't accidentally fall to that namespace.